core/src/main/java/software/amazon/awssdk/core/http/SdkHttpMetadata.java

```diff
@@ -25,7 +25,7 @@
  */
 @Immutable
 @SdkPublicApi
-public class SdkHttpMetadata {
+public final class SdkHttpMetadata {
```

Comment:
This is more of `SdkHttpData`. It's not really metadata, to me at least.

Headers are currently an attribute directly at the `SdkRequest` level. Should we add an extra level of indirection for Requests as well? `SdkHttpRequestMetadata`?

Reply:
Fine with `SdkHttpData`, but we are using `SdkHttpMetadata` in v1 though. Either way is fine with me.